spec/unit/provider/zone/solaris_spec.rb

```diff
@@ -149,6 +149,25 @@
         })
       end
     end
+    describe "with an invalid or unrecognized config" do
```

Comment:
@MikaelSmith here's my first take at a spec-based testing solution. The wording is possibly somewhat off (invalid, erroneous, unrecognized), but it covers the `provider.err` and `provider.debug` cases.